src/stage_model.py

```diff
@@ -41,19 +41,37 @@ def add_tokenizer(artifact_dir):
     tokenizer = AutoTokenizer.from_pretrained("answerdotai/ModernBERT-base")
     tokenizer.save_pretrained(artifact_dir)
 
+def add_custom_inference_script(artifact_dir):
+    """Add custom inference script to the artifact directory.
+    Args:
+        artifact_dir (str): Path to the artifact directory.
+    """
+    code_dir = os.path.join(artifact_dir, "code")
+    os.makedirs(code_dir, exist_ok=True)
+
+    # Read the contents of your existing inference.py
+    with open("src/inference.py", "r") as src_file:
```

Comment:
Feel free to leave it as is, but just to note this will assume you're in your main directory. Alternatively, if you want something more generic you could use something like
```
dirname = os.path.dirname(__file__)
filename = os.path.join(dirname, 'inference.py')
```

Reply:
nice